com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -480,97 +483,127 @@ static public void addToPaths(String dir) {
         addToPath("java.library.path", dir);
     }
 
-    private static void addOption(Options options, String shortOpt, String longOpt, boolean hasArg, String description, boolean usedByResourceCacheKey) {
-        options.addOption(shortOpt, longOpt, hasArg, description);
+    public static final class CommandLineOption {
+        public final String shortOpt;
+        public final String longOpt;
+        public final ArgCount argCount;
+        public final ArgType argType;
+        public final String description;
+
+        public CommandLineOption(String shortOpt, String longOpt, ArgCount argCount, ArgType argType, String description) {
+            this.shortOpt = shortOpt;
+            this.longOpt = longOpt;
+            this.argCount = argCount;
+            this.argType = argType;
+            this.description = description;
+        }
+
+        public enum ArgCount {ZERO, ONE, MANY}
+
+        public enum ArgType {UNSPECIFIED, ABS_OR_CWD_REL_PATH}
+    }
+
+    private static CommandLineOption opt(String shortOpt, String longOpt, CommandLineOption.ArgCount argCount, CommandLineOption.ArgType argType, String description,
+                                         boolean usedByResourceCacheKey) {
         if (usedByResourceCacheKey) {
             ResourceCacheKey.includeOption(longOpt);
         }
+        if (argCount == MANY) {
+            description = description + ". More than one occurrence is allowed";
+        }
+        return new CommandLineOption(shortOpt, longOpt, argCount, argType, description);
+    };
+
+    private static CommandLineOption opt(String shortOpt, String longOpt, CommandLineOption.ArgCount argCount, String description, boolean usedByResourceCacheKey) {
+        return opt(shortOpt, longOpt, argCount, CommandLineOption.ArgType.UNSPECIFIED, description, usedByResourceCacheKey);
     }
 
-    public static Options getCommandLineOptions() {
-        Options options = new Options();
-        addOption(options, "r", "root", true, "Build root directory. Default is current directory", true);
-        addOption(options, "o", "output", true, "Output directory. Default is \"build/default\"", false);
-        addOption(options, "i", "input", true, "DEPRECATED! Use --root instead", true);
-        addOption(options, "v", "verbose", false, "Verbose output", false);
-        addOption(options, "h", "help", false, "This help message", false);
-        addOption(options, "a", "archive", false, "Build archive", false);
-        addOption(options, "ea", "exclude-archive", false, "Exclude resource archives from application bundle. Use this to create an empty Defold application for use as a build target", false);
-        addOption(options, "e", "email", true, "User email", false);
-        addOption(options, "u", "auth", true, "User auth token", false);
+    public static List<CommandLineOption> getCommandLineOptions() {
```

Comment:
Some options accept relative paths. The paths are resolved against the JVM's working directory used to launch Bob. We use the same options from the editor, but we want to resolve relative paths against the opened project root instead — the working directory of the editor process is irrelevant. I changed option definitions to include the option metadata needed to support this.